jmclient/jmclient/yieldgenerator.py

```diff
@@ -93,23 +91,24 @@ def create_my_orders(self):
 
     def oid_to_order(self, offer, amount):
         total_amount = amount + offer["txfee"]
-        mix_balance = self.wallet.get_balance_by_mixdepth()
-        max_mix = max(mix_balance, key=mix_balance.get)
+        mix_balance = self.get_available_mixdepths()
 
-        filtered_mix_balance = [m
-                                for m in iteritems(mix_balance)
-                                if m[1] >= total_amount]
+        filtered_mix_balance = {m: b
```

Comment:
`filtered_mix_balance` could be moved to `select_input_mixdepth`. Seems cleaner to me.

Reply:
The reason it is not there is that filtering for available mixdepths is a generic task, while selecting the one to use from them is something that may depend on the custom algorithm.  And splitting the generic things from the actual selection logic is the main goal of this change, thus I think we should keep as much generally useful logic in `oid_to_order` as possible.  (Or put another way, `select_input_mixdepth` should need to do as little as possible, as that is what custom algorithms will override and reimplement.)